packages/app-desktop/gui/NoteEditor/NoteBody/PlainEditor/PlainEditor.tsx

```diff
@@ -1,14 +1,76 @@
-
-// Used in safe mode
-
 import * as React from 'react';
 import { ForwardedRef } from 'react';
 import { useEffect, useCallback, useRef, forwardRef, useImperativeHandle } from 'react';
 import { NoteBodyEditorProps, NoteBodyEditorRef } from '../../utils/types';
+const { clipboard } = require('electron');
 
 const PlainEditor = (props: NoteBodyEditorProps, ref: ForwardedRef<NoteBodyEditorRef>) => {
 	const editorRef = useRef<HTMLTextAreaElement>();
```

Comment:
Please include a brief description of how these changes fix the issue.

At present, the `PlainEditor` should only be used in safe mode ([it's initialized here](https://github.com/laurent22/joplin/blob/24954bd0f00f310e9ce233912f175017a84f0e21/packages/app-desktop/gui/NoteEditor/NoteEditor.tsx#L436)). See [TinyMCE.tsx](https://github.com/laurent22/joplin/blob/dev/packages/app-desktop/gui/NoteEditor/NoteBody/TinyMCE/TinyMCE.tsx) for the Rich Text Editor code.

Thank you for working on this!

Reply:
Hii just added the new commit, turns out there was a problem in ```onCut``` function in ```copyHtmlToClipboard(selectedContent);```
when I removed it I found out that on pasting after a cut I was getting one extra square bracket something like this
- [ ] []
test
- [ ] test2

and I found out that the if statement mention below from ```processPastedHtml``` function in resourceHandling.ts was adding that extra square bracket and a line brake so I  explicitly added an extra check such that it doesn't execute when we are pasting a checklist 
```
	if (htmlToMd && mdToHtml) {
		const md = await htmlToMd(MarkupLanguage.Markdown, html, '');
		html = (await mdToHtml(MarkupLanguage.Markdown, md, markupRenderOptions({ bodyOnly: true }))).html;

		// When plugins that add to the end of rendered content are installed, bodyOnly can
		// fail to remove the wrapping paragraph. This works around that issue by removing
		// the wrapping paragraph in more cases. See issue #10061.
		if (!md.trim().includes('\n')) {
			html = removeWrappingParagraphAndTrailingEmptyElements(html);
		}
	}
```

On making these changes I was able to cut and paste properly. The video below is demo of latest changes.

[Screencast from 2024-07-12 11-19-55.webm](https://github.com/user-attachments/assets/ef2726ee-0cbb-41c2-a8d5-796b3604fc8c)


While raising the PR earlier the cut past worked on video because I was mistakenly performing copy instead of cut😅